### PR TITLE
fix(YouTube/Sponsorblock) Fix import json settings.  Fix default userid showing as empty.  Fix skip segment toast not working.

### PR DESCRIPTION
**Pull request by LisoUseInAIKyrios**
- fixed importation of browser json settings (removed importation of non standard SB_LAST_VIP_CHECK)
- fixed user id always showing as empty string (prevented voting and user stats from working)
- fixed skip segment toast not working (settings activity was not passing some values into SettingEnum)

Everything appears to be working now.

Fixes revanced/revanced-patches/issues/1105, fixes revanced/revanced-patches/issues/1447.

### CODE REVIEW

File: app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java [hunk 1]
```diff
@@ -541,18 +541,16 @@ public static void importSettings(String json, Context context) {
...
             SettingsEnum.SB_MIN_DURATION.saveValue(Float.valueOf(settingsJson.getString("minDuration")));
-            SettingsEnum.SB_UUID.saveValue(settingsJson.getString("userID"));
-            SettingsEnum.SB_LAST_VIP_CHECK.saveValue(settingsJson.getLong("lastIsVipUpdate"));
-
+            SettingsEnum.SB_COUNT_SKIPS.saveValue(settingsJson.getBoolean("trackViewCount"));
 
             String serverAddress = settingsJson.getString("serverAddress");
             if (serverAddress.equalsIgnoreCase("https://sponsor.ajay.app")) {
-                serverAddress = (String) SettingsEnum.SB_API_URL.getDefaultValue();
+                serverAddress = (String) SettingsEnum.SB_API_URL.getDefaultValue(); // upgrade from old url
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java:553]: What do you mean by that?

File: app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java [hunk 2]
```diff
@@ -585,16 +583,15 @@ public static String exportSettings(Context context) {
...
+            json.put("serverAddress", SettingsEnum.SB_API_URL.getString());
             json.put("dontShowNotice", !SettingsEnum.SB_SHOW_TOAST_WHEN_SKIP.getBoolean());
-            json.put("barTypes", barTypesObject);
             json.put("showTimeWithSkips", SettingsEnum.SB_SHOW_TIME_WITHOUT_SEGMENTS.getBoolean());
             json.put("minDuration", SettingsEnum.SB_MIN_DURATION.getFloat());
             json.put("trackViewCount", SettingsEnum.SB_COUNT_SKIPS.getBoolean());
             json.put("categorySelections", categorySelectionsArray);
-            json.put("userID", SettingsEnum.SB_UUID.getString());
-            json.put("isVip", SettingsEnum.SB_IS_VIP.getBoolean());
-            json.put("lastIsVipUpdate", SettingsEnum.SB_LAST_VIP_CHECK.getLong());
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java:596]: This setting is missing now. What is the motivation?

File: app/src/main/java/app/revanced/integrations/settingsmenu/ReturnYouTubeDislikeSettingsFragment.java [hunk 3]
```diff
@@ -62,8 +62,6 @@ public void onCreate(Bundle savedInstanceState) {
         setPreferenceScreen(preferenceScreen);
 
         enabledPreference = new SwitchPreference(context);
-        enabledPreference.setKey(SettingsEnum.RYD_ENABLED.getPath());
-        enabledPreference.setDefaultValue(SettingsEnum.RYD_ENABLED.getDefaultValue());
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/integrations/settingsmenu/ReturnYouTubeDislikeSettingsFragment.java:66]: Why was this removed?

File: app/src/main/java/app/revanced/integrations/settings/SettingsEnum.java [hunk 4]
```diff
@@ -366,6 +366,7 @@ public void saveValue(Object newValue) {
         value = newValue;
     }
 
+    // FIXME?  why does this return a primitive, when the other methods return objects?  Should this return an Integer?
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/integrations/settings/SettingsEnum.java:369]: This was likely inherited from reversed Vanced YouTube code. Using the primitive equivalent for all methods should be sufficient.

File: app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java (hunk 1 above)
Comment (LisoUseInAIKyrios) [reply app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java:553]: the logic was already there.  The browser extension gives the url 'https://sponsor.ajay.app'
but ReVanced integration wants to use "https://sponsor.ajay.app/api/"

I'll clarify the comment.

File: app/src/main/java/app/revanced/integrations/settingsmenu/ReturnYouTubeDislikeSettingsFragment.java (hunk 3 above)
Comment (LisoUseInAIKyrios) [reply app/src/main/java/app/revanced/integrations/settingsmenu/ReturnYouTubeDislikeSettingsFragment.java:66]: because it's does nothing, and the code gives the impression it does something.

File: app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java (hunk 1 above)
Comment (oSumAtrIX) [reply app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java]: Shouldn't we migrate to `https://sponsor.ajay.app/` then?

File: app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java (hunk 2 above)
Comment (LisoUseInAIKyrios) [reply app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java:596]: Only SB_LAST_VIP_CHECK was removed (it's not a setting to import/export).

The other fields were rearranged, so the json export better matches the appearance of the browser extension.

Comment (oSumAtrIX) [reply app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java:596]: Is the setting `SB_LAST_VIP_CHECK` relevant elsewhere in the code?

File: app/src/main/java/app/revanced/integrations/settings/SettingsEnum.java (hunk 4 above)
Comment (LisoUseInAIKyrios) [reply app/src/main/java/app/revanced/integrations/settings/SettingsEnum.java]: primitive return types would be more clear, since null number values cannot be saved anyways

File: app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java (hunk 2 above)
Comment (LisoUseInAIKyrios) [reply app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java:596]: it's only a date of the last time vip status was checked.  It's only used to prevent checking for vip status all the time, and it's not relevant for importing or exporting (the browser extension does import or export this field)

File: app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java (hunk 1 above)
Comment (LisoUseInAIKyrios) [reply app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java]: yeah probably.  But my goal here is just to fix the bugs.

Comment (LisoUseInAIKyrios) [reply app/src/main/java/app/revanced/integrations/sponsorblock/SponsorBlockUtils.java]: I just checked again, and the browser extension does not accept importing "http://server/api/"

I changed ReVance to behave the same, and it import/exports only the host name ("http://server")

**Comment by LisoUseInAIKyrios** [comment]
should be good to go

File: app/src/main/java/app/revanced/integrations/sponsorblock/requests/SBRequester.java [hunk 5]
```diff
@@ -129,12 +130,13 @@ public static void sendViewCountRequest(SponsorSegment segment) {
...
 
     public static void voteForSegment(SponsorSegment segment, VoteOption voteOption, Context context, String... args) {
-        new Thread(() -> {
+        ReVancedUtils.runOnBackgroundThread(() -> {
             try {
                 String segmentUuid = segment.UUID;
                 String uuid = SettingsEnum.SB_UUID.getString();
                 String vote = Integer.toString(voteOption == VoteOption.UPVOTE ? 1 : 0);
 
+                // edit: could probably remove this toast, as another toast with the result is shown only a few seconds later
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/integrations/sponsorblock/requests/SBRequester.java:139]: Can you elaborate on this?

Comment (oSumAtrIX) [reply app/src/main/java/app/revanced/integrations/sponsorblock/requests/SBRequester.java:140]: That makes sense which is why it should be removed.

**Review by oSumAtrIX — APPROVED** [review]
